include/alpaka/pltf/PltfGenericSycl.hpp

```diff
@@ -18,89 +18,150 @@
 
 #    include <CL/sycl.hpp>
 
-#    include <iostream>
+#    include <cstddef>
+#    include <exception>
+#    include <functional>
+#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+#        include <iostream>
+#    endif
+#    include <mutex>
+#    include <optional>
+#    include <shared_mutex>
 #    include <sstream>
 #    include <stdexcept>
 #    include <vector>
 
 namespace alpaka::experimental
 {
     //! The SYCL device manager.
-    class PltfGenericSycl : public concepts::Implements<ConceptPltf, PltfGenericSycl>
+    template<typename TSelector>
+    class PltfGenericSycl : public concepts::Implements<ConceptPltf, PltfGenericSycl<TSelector>>
     {
     public:
         PltfGenericSycl() = delete;
+
+        [[nodiscard]] static auto syclPlatform() -> sycl::platform&
+        {
+            auto lock = std::unique_lock<std::mutex>{platform_mutex};
+
+            if(!platform_opt.has_value())
```

Comment:
Optional is Boolish (i.e. contextually convertible to bool), like pointers. So it is often used like this:
```suggestion
            if(!platform_opt)
```

Reply:
Yes, I know. I wanted to make this explicit.

Reply:
That's what I thought. Fine then.